src/porepy/grids/boundary_grid.py

```diff
@@ -56,24 +51,50 @@ def __init__(self, g: pp.Grid, name: Optional[str] = None) -> None:
         self._parent = g
         """Parent grid from which the boundary grid is constructed."""
 
-        self.cell_centers = g.face_centers[:, parent_boundary]
+        if g.dim == 0:
+            raise ValueError("Boundary grids are not supported for 0d grids.")
+        self.dim = g.dim - 1
+        """Dimension of the boundary grid."""
+
+        self.num_cells: int
+        """Number of cells in the boundary grid.
+
+        Will correspond to the number of faces on the domain boundary in the parent
+        grid. Initialized in :meth:`~compute_geometry`.
+
+        """
```

Comment:
I'm tempted to suggest self._projections

Reply:
Ok